src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -0,0 +1,377 @@
+                    FileMode fileMode = overwrite ? FileMode.Create : FileMode.CreateNew;
+
+                    using (FileStream fs = new(destinationFileName, fileMode, FileAccess.Write, FileShare.None))
+                    {
+                        if (DataStream != null)
+                        {
+                            if (DataStream.CanSeek)
+                            {
+                                // Make sure to rewind the data stream in case it was opened and read externally before calling this method.
+                                DataStream.Seek(0, SeekOrigin.Begin);
+                            }
+                            DataStream.CopyTo(fs);
+                            SetModeOnFile(fs.SafeFileHandle, destinationFileName);
+                        }
+                    }
+
+                    try
+                    {
+                        File.SetLastWriteTime(destinationFileName, ModificationTime.DateTime);
+                    }
+                    catch (UnauthorizedAccessException)
+                    {
+                        // some OSes like Android (#35374) might not support setting the last write time, the extraction should not fail because of that
+                    }
+
+                    break;
+
+                case TarEntryType.SymbolicLink:
+                    FileInfo link = new(destinationFileName);
+                    link.CreateAsSymbolicLink(LinkName);
+                    break;
+
+                case TarEntryType.HardLink:
+                    ExtractAsHardLink(destinationFileName);
+                    break;
+
+                case TarEntryType.BlockDevice:
+                    ExtractAsBlockDevice(destinationFileName);
+                    break;
+
+                case TarEntryType.CharacterDevice:
+                    ExtractAsCharacterDevice(destinationFileName);
+                    break;
+
+                case TarEntryType.Fifo:
+                    ExtractAsFifo(destinationFileName);
+                    break;
+
+                case TarEntryType.ExtendedAttributes:
+                case TarEntryType.GlobalExtendedAttributes:
+                case TarEntryType.LongPath:
+                case TarEntryType.LongLink:
+                    Debug.Assert(false, $"Metadata entry type should not be visible: '{EntryType}'");
+                    break;
+
+                case TarEntryType.MultiVolume:
+                case TarEntryType.RenamedOrSymlinked:
+                case TarEntryType.SparseFile:
+                case TarEntryType.TapeVolume:
+                default:
+                    throw new NotSupportedException(string.Format(SR.TarEntryTypeNotSupportedForExtracting, EntryType));
+            }
+        }
+
+        public Task ExtractToFileAsync(string destinationFileName, bool overwrite, CancellationToken cancellationToken = default)
+        {
+            throw new NotImplementedException();
```

Comment:
I assume it's on the TODO list?

Reply:
Yes, async will have to be implamented later.

Reply:
If we're not going to implement these as part of this PR, I'd prefer they be removed, at least from the ref, until they're implemented.  It's otherwise a bad experience to add new APIs that aren't yet implemented.

Reply:
No problem. Removing them from the ref.